### DISCUSSION TITLE
SPI-API URI construction & Signing in VBA - eek

### DISCUSSION THREAD

**Discussion by peskywinnets** [amzn/selling-partner-api-models#3597] [General] [answered]
I seem to be walking a solitary path here.

I'm a one man business who using MS Access to help me process orders...I'm not really a programmer, but managed to learn enough VBA to kludge together an API using VBA for the old (soon to be defunct) MW API ..but the signing of the new SP-API is on another level of complexity.

I've managed to create all the IAM user, role & policy, I've self authorize my app in seller central...I've used my LWA credentials to retrieve an Access token successfully from https://api.amazon.com/auth/o2/token , but constructing a simple URI & signing it with the access token...defeats me.

Anyone here managed it, or could give me pointers as to how I might see how to construct the URI & sign it this using other tools? (I have all the encryption modules I need in VBA I believe, it's just the order of things I'm struggling with)

Many thanks in anticipation

**Comment by intellisofts** [comment]
Hi peskywinnets,
I have automated a lot of APIs in MS Access VBA with OAuth 2.0. Fortunately I am working on New SP API and finding the sandbox credentials to work on while I reached here. I can assist you with Access Token and calling endpoints using it. I do have state of the art JSON Parser for VBA.
I'll look forward your early response.
Kind regards,
Farrukh Rehman
CEO Intelligent Software Solutions

**Comment by MeenaAmz** [comment, marked as answer]
@peskywinnets SP-API does not require AWS signing anymore. See more details [here](https://github.com/amzn/selling-partner-api-docs/discussions/3385).

**Comment by QasimMughal05** [comment]
@intellisofts,@MeenaAmz
Are you working on SP API + VBA? Please Let me know

**Reply by intellisofts** [reply]
Yes I am working on it.

**Comment by QasimMughal05** [comment]
Does it support sp api ? And how much you have done on it?

**Reply by intellisofts** [reply]
Yes it supports. Just waiting for App to be approved and it will start working with all endpoints.

**Comment by QasimMughal05** [comment]
Can you share a brief on how it works?
